### PR TITLE
[ADD] active bullets links

### PR DESCRIPTION
**Pull request by zainab-oraby**

### CODE REVIEW

File: zee_slider.js [hunk 1]
```diff
@@ -121,9 +124,9 @@
       this.bulletsContainer.className = "zeeSlider-bullets-container";
       for (i=0; i < this.imagesCount; i++) {
         if ( i === 0 ) {
-          this.bulletsContainer.innerHTML +=  "<span class='active bullet_"+ i +"'>o</span>";
+          this.bulletsContainer.innerHTML +=  "<span class='active' data-bullet-index='"+ i +"'>o</span>";
```
Comment (Mohamed-amin) [inline zee_slider.js:127]: this is a bit less effecient, as innerHTML is very cpu-expensive.
I would suggest to one thing... 
create a `documentFragment`, give it a class `zeeSlider-bullets-container` then use `document.creatElement` and creat bullets and append to the container.

then replace the docmentFragment with the current container in the dom. 
with that you gained to major benefits
1. didn't use innerHTML :) 
2. only inserted in the dom once, which basically better than appending into it several times in loop.

File: zee_slider.js [hunk 2]
```diff
@@ -168,20 +171,45 @@
     }
   }
 
+  function initializeBulletsLinks() {
+    var _ = this,
```
Comment (Mohamed-amin) [inline zee_slider.js:175]: thats can be avoided by number of techniques, 
for starter you can use `arrow functions`  you won't have to do this funny stuff of assigning this. 
especially use it on the foreach callback if you want to access the outer function scope easily.

File: zee_slider.js [hunk 3]
```diff
@@ -168,20 +171,45 @@
...
-    this.currentBullet = this.bulletsContainer.querySelector('span.bullet_' + this.currentImgIndex);
+    this.currentBullet = this.bulletsContainer.querySelector('[data-bullet-index="' + this.currentImgIndex + '"]');
     this.currentBullet.className += ' active';
   }
 
-  function playImageAnimation() {
-    this.currentImage = document.querySelector('[data-img-index="' + this.currentImgIndex + '"]');
-    this.currentImage.style.display = 'block';
+  function playImageAnimation( imageIndex ) {
+    if ( typeof imageIndex !== 'undefined' ) {
```
Comment (Mohamed-amin) [inline zee_slider.js:206]: you don't neet to check for the type.
just do 
```js
imageIndex !== undefined // undefined without ' ' quotations
```

File: zee_slider.js [hunk 4]
```diff
@@ -56,6 +56,9 @@
     if (this.options.autoPlay === true) {
       initializeAutoPlay.call(this);
     }
+    if (this.options.showBullets === true) {
+      initializeBulletsLinks.call(this);
```
Comment (Mohamed-amin) [inline zee_slider.js:60]: using call here is a bit wired, we can design the class like 
```
const slider =  {
  index: 0,
current:1,
blah: 'sss',
get: function(){
return this.index; /// here you get access to the parent object.
}
}
```

or 
```js
function Slider(){
  this.name = 'zienab';
  this.get = function(){
	return this.name;
  }
}
var slider = new Slider();
slider.get() // zienab
```

**Comment by Mohamed-amin** [comment]
zZzZzZzZzZzZ 😴  @zmmorabya

**Comment by zainab-oraby** [comment]
@Mohamed-amin And I was wondering why you were inactive and not responding to the pull request :D I just saw all the comments in the email notification right before this comment. :D 
I will check them tomorrow and respond to each one. 👍

File: zee_slider.js (hunk 1 above)
Comment (zainab-oraby) [reply zee_slider.js:127]: @Mohamed-amin This is what the buildOut method does. It creates all elements of the slider in a docFrag and after that happens, the docFrag is appended to the DOM.
Do you mean even inner elements in the docFrag such as the bullets should be added in another docFrag as well, then appended to the parent docFrag?

File: zee_slider.js (hunk 4 above)
Comment (zainab-oraby) [reply zee_slider.js:60]: @Mohamed-amin  I didn't get this part. :confused: 
And it's Zainab not otherwise.  :grinning: :smiley:

Comment (Mohamed-amin) [reply zee_slider.js:60]:
```js
initializeBulletsLinks()
```
this func depends on other members of the slider class.
so you can make it part of the class it self.

File: zee_slider.js (hunk 1 above)
Comment (Mohamed-amin) [reply zee_slider.js:127]: oh, sorry my bad I didn't see the above code :)

File: zee_slider.js (hunk 4 above)
Comment (zainab-oraby) [reply zee_slider.js:60]: @Mohamed-amin Still not there yet. I will give it a try and push it to a separate commit.

**Pull request merged**